libs/utils/android/workloads/viewer.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+import logging
+
+from time import sleep
+
+from android import Screen, System
+from android.workload import Workload
+
+
+class Viewer(Workload):
+    """
+    Android generic Viewer workload
+    """
+    
+    # Let the system pick the best package
+    package = ''
+
+    def __init__(self, test_env):
+        super(Viewer, self).__init__(test_env)
+
+        # Set of output data reported by the viewer
+        self.db_file = None
+
+    def pre_viewing(self):
```

Comment:
Yeah it's mostly to surround the tracing calls. The pre-viewing is there mostly for video playback, as sometimes the VIEW intent is not enough and you need to press play or something before properly "viewing" the video. Since the `viewing()` method is there for the actual workload (swiping, tapping, waiting...), it's then obvious that `pre_viewing()` and `post_tracing()` are executed just before/after the workload itself. `setup/teardown` works fine too.